membrane_curvature/base.py

```diff
@@ -0,0 +1,126 @@
+"""
+MembraneCurvature
+=======================================
+
+:Author: Estefania Barreto-Ojeda
+:Year: 2021
+:Copyright: GNU Public License v3
+
+MembraneCurvature calculates the mean and Gaussian curvature of
+surfaces derived from a selection of reference.
+"""
+
+import numpy as np
+import warnings
+from membrane_curvature.surface import get_z_surface
+from membrane_curvature.curvature import mean_curvature, gaussian_curvature
+
+
+from MDAnalysis.analysis.base import AnalysisBase
+import MDAnalysis as mda
+
+
+class MembraneCurvature(AnalysisBase):
+
+    r"""
+    MembraneCurvature is a tool to calculate membrane curvature.
+
+    Parameters
+    ----------
+    universe : Universe or AtomGroup
+        An MDAnalysis Universe object.
+    select : str or iterable of str, optional. Default: 'all'
+        The selection string of an atom selection to use as a
+        reference to derive a surface.
+    pbc : bool, optional. Default: 'True'
+        Apply periodic boundary conditions.
+    n_x_bins : int, optional. Deafult: '100'
+        Number of bins in grid in the `x` dimension.
+    n_y_bins : int, optional. Default: '100'
+        Number of bins in grid in the `y` dimension.
+    x_range : tuple of (float, float), optional. Deafult: (0, `universe.dimensions[0]`)
+        Range of coordinates (min, max) in the `x` dimension.
+    y_range : tuple of (float, float), optional. Deafult: (0, `universe.dimensions[1]`)
+        Range of coordinates (min, max) in the `y` dimension.
+
+    Attributes
+    ----------
+    results.z_surface : ndarray
+        Surface derived from atom selection in every frame.
+        Array of shape (`n_frames`, `n_x_bins`, `n_y_bins`)
+    results.mean_curvature : ndarray
+        Mean curvature associated to the surface.
+        Array of shape (`n_frames`, `n_x_bins`, `n_y_bins`)
+    results.gaussian_curvature : ndarray
+        Gaussian curvature associated to the surface.
+        Arrays of shape (`n_frames`, `n_x_bins`, `n_y_bins`)
+
+    Returns
+    -------
+    average_z_surface : ndarray
+        Average of the array elements in `z_surface`
+    average_mean_curvature :
+        Average of the array elements in `mean_curvature`
+    average_gaussian_curvature:
+        Average of the array elements in `gaussian_curvature`
```

Comment:
It does not make sense to specify return values for a class unless the class is pretending to be a function (with `__call__`) but that's not what Analysis classes do. They store results in `results` after executing `run()` — exactly as you documented. The `__init__` method does not have a return value.

Remove the *Returns* section. Add whatever `_conclude()` adds to `results` to the attribute section. Add some text explaining that these specific attributes are available after run.